libs/MeshKernel/include/MeshKernel/Mesh.hpp

```diff
@@ -122,6 +122,10 @@ namespace meshkernel
         /// @brief Default constructor
         Mesh() = default;
 
+        /// @brief  Constructs an empty mesh, sets only the projection
+        /// @param[in] projection  The projection to use
+        Mesh(Projection projection) : m_projection(projection){}
```

Comment:
Why not simply have a public method to set the projection in the base class Mesh? Something like
```
 Mesh::SetProjection(Projection projection) {  m_projection = projection; }
```
This saves you defining custom constructor in Mesh and all its derived classes: Mesh1D, Mesh2D and CurvilinearGrid.
Do you think this is possible?

Reply:
i think is better to set the projection in the constructor even if this requires a bit more code